lib/Backend/Commands.hs

```diff
@@ -87,7 +88,27 @@ createCmd (CreateOptions entryPath _edit force tags fields privateFields) = do
       & E.fields .~ HashMap.fromList allFields
       & E.tags .~ tags
 
-  -- TODO: check that a directory does not exist at the given path.
+    checkForEntriesInEntryPath :: Member BackendEffect r => EntryPath -> Sem r (Maybe EntryPath)
+    checkForEntriesInEntryPath entryPath =
+      let 
+        Path.EntryPath (start NE.:| rest) = entryPath
+        startPath = Path.EntryPath (start NE.:| [])
+        restPath = Path.Path rest
+      in process startPath restPath
+        where 
+          process :: Member BackendEffect r => EntryPath -> Path -> Sem r (Maybe EntryPath)
+          process _ (Path.Path []) = pure Nothing
+          process entryPath (Path.Path (x : xs)) = 
+            pathIsEntry entryPath >>= \case
+              True -> (pure . Just) entryPath
```

Comment:
Instead of returning the entry here, and then using `when isJust + throw` in line 110, we could simplify this by throwing the error here.